src/NumberField/NumberField.js

```diff
@@ -20,20 +20,53 @@ export default function NumberField({
   value,
   variant = "outlined"
 }) {
+  const [valueError, setValueError] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState("");
+
+  // get max and min values from inputProps
+  let max;
```

Comment:
It might be cleaner and more obvious from a documentation point of view that the feature exists if min, max and step were their own separate props rather than inputProps. Maybe also a prop to disable the stepper would be good too.

Reply:
Ah you already have a prop to disable the stepper. I would go all the way and do the same for min, max and step then.

Reply:
Should I remove inputProps as a prop in the process?

Reply:
I would unless you can think of a reason to keep it.